### PR TITLE
CIV-14008 GA Lip Hwf

### PR DESCRIPTION
**Pull request by kalachandrasekar1**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-14008

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/common/models/generalApplication/GeneralApplication.ts [hunk 1]
```diff
@@ -43,6 +49,10 @@ export class GeneralApplication {
     response?: GaResponse,
     uploadEvidenceForApplication?: UploadGAFiles,
     statementOfTruth?: StatementOfTruthForm,
+    applyHelpWithFees?: YesNo,
```
Comment (neeta-hmcts) [inline src/main/common/models/generalApplication/GeneralApplication.ts:52]: can we create a model for HWF related fields.

File: src/main/modules/i18n/locales/cy.json [hunk 2]
```diff
@@ -2494,6 +2494,24 @@
         "SINGLE_DATE": "A single date",
         "TITLE": "Are there any dates when you cannot attend a hearing within the next 3 months (optional)?",
         "TO": "To"
+      },
+      "APPLY_HELP_WITH_FEE": {
+        "TITLE": "Pay application fee",
```
Comment (neeta-hmcts) [inline src/main/modules/i18n/locales/cy.json:2499]: arrange the strings in asc order

File: src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeContinueController.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const claim = await getClaimById(claimId, <AppRequest>req, true);
+    const lng = req.query.lang ? req.query.lang : req.cookies.lang;
+    const option = req.body.option;
+    const form = new GenericForm(new GenericYesNo(option, t('ERRORS.VALID_YES_NO_SELECTION_UPPER', { lng })));
+    await form.validate();
+    if (form.hasErrors()) {
+      await renderView(res, req, form, claimId, lng);
+    } else {
+      let redirectUrl;
+      if (req.body.option == YesNo.YES) {
```
Comment (neeta-hmcts) [inline src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeContinueController.ts:67]: use === instead == and can use ternary operator here

File: src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeController.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import {AppRequest} from 'models/AppRequest';
+import {
+  getApplyHelpWithApplicationFeeContents, getButtonsContents,
+} from 'services/features/generalApplication/applicationFee/helpWithApplicationFeeContent';
+import {saveHelpApplicationFeeSelection} from 'services/features/generalApplication/generalApplicationService';
+import {generateRedisKey} from 'modules/draft-store/draftStoreService';
+
+const applyHelpWithApplicationFeeViewPath  = 'features/generalApplication/applicationFee/help-with-application-fee';
+const helpWithApplicationFeeController = Router();
+const backLinkUrl = 'test'; // TODO: add url
```
Comment (neeta-hmcts) [inline src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeController.ts:19]: add backlink url too if possible.

File: src/main/routes/features/generalApplication/applicationFee/helpWithFeesContentController.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      applyHelpWithFeeContents:getHelpWithApplicationFeeContent(claimId),
+    });
+}) as RequestHandler);
+
+helpWithFeesContentController.post(GA_APPLY_HELP_WITH_FEES_START, (async (req: AppRequest | Request, res: Response) => {
+  const claimId = req.params.id;
+  res.redirect(constructResponseUrlWithIdParams(claimId, GA_APPLY_HELP_WITH_FEE_REFERENCE));
+})as RequestHandler);
+
+const getHelpWithApplicationFeeContent = (claimId: string) => {
```
Comment (neeta-hmcts) [inline src/main/routes/features/generalApplication/applicationFee/helpWithFeesContentController.ts:32]: this function can be moved outside controller

**Review by jarekPierchala — APPROVED** [review]

File: src/main/routes/features/generalApplication/applicationFee/helpWithFeesContentController.ts (hunk 5 above)
Comment (kalachandrasekar1) [reply src/main/routes/features/generalApplication/applicationFee/helpWithFeesContentController.ts]: moved out of controller

File: src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeContinueController.ts (hunk 3 above)
Comment (kalachandrasekar1) [reply src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeContinueController.ts]: updated

File: src/main/common/models/generalApplication/GeneralApplication.ts (hunk 1 above)
Comment (kalachandrasekar1) [reply src/main/common/models/generalApplication/GeneralApplication.ts]: created new model

**Review by jarekPierchala — APPROVED** [review]

**Comment by jarekPierchala** [comment]
part of ga_lip_bulk branch

**Pull request closed without merging**